wiki/FitNesseRoot/HsacAcceptanceTests/SlimTests/BrowserTest/ExecuteScript.wiki

```diff
@@ -0,0 +1,29 @@
+---
+Test
+---
+This test ensures that we can execute javascript on a page.
+
+We use a mock server running to serve the HTML defined in this page.
+
+
+!define HTML { {{{
+<html>
+<body>
+<input type="date" id="dateField">
+</body>
+</html>}}} }
+
+
+|script      |mock xml server setup|
+|add response|${HTML}              |
+|$url=       |get mock server url  |
+
+|script                |browser test                                                    |
+|seconds before timeout|1                                                               |
+|open                  |$url                                                            |
+|execute script        |document.getElementById("dateField").setAttribute("type","text")|
+|check                 |value of attribute    |type    |on    |id=dateField    |text    |
```

Comment:
Presumably in your real tests you do an `enter <value> as id=datefield` after changing the type? Is there a single date format that must always be used, or is that browser/user locale dependent?

Would it be a good idea to add (as a separate PR) a new BrowserTest command to select a date. The java method would set the attribute type (on the element found using browser test normal heuristic, passing it as `arguments[0]`), enter the value, and change the attribute back?